Shared/Samples/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesView.Model.swift

```diff
@@ -0,0 +1,273 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+import ExternalAccessory
+
+extension DisplayDeviceLocationWithNMEADataSourcesView {
+    /// The model used to store the geo model and other expensive objects used in this view.
+    class Model: ObservableObject {
+        /// A map with a navigation basemap.
+        let map = Map(basemapStyle: .arcGISNavigation)
+        
+        /// The current autopan mode.
+        var autoPanMode: LocationDisplay.AutoPanMode {
+            get {
+                locationDisplay.autoPanMode
+            }
+            set {
+                locationDisplay.autoPanMode = newValue
+            }
+        }
+        
+        /// A Boolean value specifying if the "recenter" button should be disabled.
+        @Published var isRecenterButtonDisabled = true
+        
+        /// A Boolean value specifying if the "reset" button should be disabled.
+        @Published var isResetButtonDisabled = true
+        
+        /// A Boolean value specifying if the "source" button should be disabled.
+        @Published var isSourceButtonDisabled = false
+        
+        /// A string containing GPS accuracy.
+        @Published var accuracyStatus: String = "Accuracy info will be shown here."
+        
+        /// A string containing satellite information.
+        @Published var satelliteStatus: String = "Satellites info will be shown here."
+        
+        /// The location display used in the map view.
+        var locationDisplay: LocationDisplay = {
+            let locationDisplay = LocationDisplay()
+            locationDisplay.autoPanMode = .recenter
+            return locationDisplay
+        }()
+        
+        /// An NMEA location data source, to parse NMEA data.
+        var nmeaLocationDataSource: NMEALocationDataSource!
```

Comment:
```swift
        private var nmeaLocationDataSource: NMEALocationDataSource!
```
Ideally this data source should be private, and only accessible via `locationDisplay.dataSource`. More on this below.